### PR TITLE
Cleanup component level custom properties

### PR DESCRIPTION
**Pull request by kyledurand**
Part of https://github.com/Shopify/polaris-react/issues/4732

This changes all properties created at component levels to the `--pc-<component>` prefix convention. I did this fairly quickly so it's probably going to be a bit broken. Don't feel like you need to review straight away

### CODE REVIEW

File: src/components/ActionList/ActionList.scss [hunk 1]
```diff
@@ -31,13 +31,10 @@ $item-vertical-padding: ($item-min-height - line-height(body)) / 2;
...
   @include text-style-subheading;
-
-  --p-titleVerticalSpacing: calc(var(--p-space-2) * 1.5);
-  padding: var(--p-space-1) var(--p-space-4) var(--p-titleVerticalSpacing)
-    var(--p-space-4);
+  padding: var(--p-space-1) var(--p-space-4) var(space-3) var(--p-space-4);
 
   &.firstSectionWithTitle {
-    padding-top: var(--p-titleVerticalSpacing);
+    padding-top: var(space-3);
```
Comment (alex-page) [inline src/components/ActionList/ActionList.scss]:
```suggestion
    padding-top: var(--p-space-3);
```

File: src/components/ColorPicker/ColorPicker.scss [hunk 2]
```diff
@@ -13,9 +13,9 @@ $inner-shadow: inset 0 0 2px 0 rgba(0, 0, 0, 0.5);
 }
 
 .ColorPicker {
-  --pc-color: 10;
-  --pc-adjustments: 20;
-  --pc-dragger: 30;
+  --pc-color-picker-color: 10;
```
Comment (alex-page) [inline src/components/ColorPicker/ColorPicker.scss]: Maybe this should be called sheet or background. It's not really `color`. Any thoughts @kyledurand ?

**Review by alex-page — APPROVED** [review]
This is going to set us up to move super fast. Thanks @kyledurand 🥇

File: src/components/ColorPicker/ColorPicker.scss (hunk 2 above)
Comment (kyledurand) [reply src/components/ColorPicker/ColorPicker.scss]: Oh yeah true. This node sets the root stacking context for this component with `position: absolute` so I'm just going to go with `--pc-color-picker`

**Pull request merged**